### PR TITLE
Bugfix/syal 1082

### PR DESCRIPTION
**Pull request by ThanhD-AVISPL**

### CODE REVIEW

**Review by MaksimR-AVISPL — APPROVED** [review]

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java [hunk 1]
```diff
@@ -1411,11 +1414,13 @@ private void roomControl(Map<String, String> stats, List<AdvancedControllablePro
 				if (i == roomIndex) {
 					continue;
 				}
-				if (cachedRooms.get(i).getName().equalsIgnoreCase(value)) {
+				if (!controllableProperty.contains(SmartThingsConstant.DELETE) && cachedRooms.get(i).getName().equalsIgnoreCase(value)) {
 					throw new ResourceNotReachableException(String.format("The room name %s already exists, Please chose the different room name", value));
```
[the remark this answers is not in the document]
Comment (ThanhD-AVISPL) [reply src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java:1418]: I fixed it

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java [hunk 2]
```diff
@@ -1529,7 +1534,10 @@ private void createRoomControl(Map<String, String> stats, List<AdvancedControlla
 				cachedCreateRoom = new Room();
 				break;
 			case CREATE_ROOM:
-				isEditedForCreateRoom = false;
+				String roomName = cachedCreateRoom.getName();
+				if (roomName == null || roomName.trim().isEmpty()){
+					throw new IllegalArgumentException("Invalid room name, the room name can not be empty");
+				}
 				if (cachedRooms.size() >= SmartThingsConstant.MAX_ROOM_QUANTITY) {
 					throw new ResourceNotReachableException(String.format("Can not create more than %s room", SmartThingsConstant.MAX_ROOM_QUANTITY));
```
[the remark this answers is not in the document]
Comment (ThanhD-AVISPL) [reply src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java:1542]: I fixed it

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java [hunk 3]
```diff
@@ -1709,6 +1717,9 @@ private void deviceDashboardControl(Map<String, String> stats, List<AdvancedCont
 			// devices control
 			try {
 				Device device = findDeviceByName(controllableProperty);
+				if (!convertDeviceStatusValue(device)) {
+					throw new ResourceNotReachableException(String.format("Unable to control %s, device is offline", device.getName()));
```
[the remark this answers is not in the document]
Comment (ThanhD-AVISPL) [reply src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java]: I fixed it

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java [hunk 4]
```diff
@@ -1556,7 +1564,7 @@ private void createRoomControl(Map<String, String> stats, List<AdvancedControlla
 					} else {
 						throw new ResourceNotReachableException(String.format("Creating room with name %s fail, please try again later", value));
```
[the remark this answers is not in the document]
Comment (ThanhD-AVISPL) [reply src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java]: I fixed it

**Review by MaksimR-AVISPL — APPROVED** [review]

**Pull request merged**